### PR TITLE
refactor: use sumiContributes and compatible with kaitianContributes 

### PR DESCRIPTION
**Pull request by erha19**
### Types

- [x] 🪚 Refactors

### <samp>🤖 Generated by Copilot at 74babfe</samp>

*  Rename the `kaitianContributes` property to `sumiContributes` in the extension package schema and the extension contribution points registration and activation logic ([link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-366289ffb7e39f53297fa56da7119b46f475882d9ac3e2ed10b03b9ba2f524e4L30-R30), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-7943476a12073afcaf3964223a391b3ac42a27789e21fb54c5531aa1112e7bdbL5-R5), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-ee02fb088a19a9a0c87b1501c7a1ce3a1a6a70ce687546360ae17f0197611417L173-R173), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-7e4610f55daeaaca635b72e1cba7d5548f4d5ad6d5e36466dc1b26052ca015c1L182-R186), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-7e4610f55daeaaca635b72e1cba7d5548f4d5ad6d5e36466dc1b26052ca015c1L193-R193), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-56c70608623e5329d4b53a62c8d82b065f5c5b4ddf3aab359a8c22b6bc354016L68-R72), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-56c70608623e5329d4b53a62c8d82b065f5c5b4ddf3aab359a8c22b6bc354016L80-R80), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-3d705c2bf9f71760b12685e3d9e61fc6339b3886d9ace8c550014b9b2547b806L274-R274), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-3d705c2bf9f71760b12685e3d9e61fc6339b3886d9ace8c550014b9b2547b806L654-R654), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-62990366c95d4e68747b72694a7f1f31eeeccd44aa455911ff445b0b3375d135L436-R439), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-62990366c95d4e68747b72694a7f1f31eeeccd44aa455911ff445b0b3375d135L518-R521), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-957921fb34847ea35a1d53cb2e5a304f3d989230afd6dd534bcc6b47b23cc4ceL212-R217))
*  Add compatibility checks for the legacy extensions that use the `kaitianContributes` property and merge it with the `sumiContributes` property in the `ExtensionScanner` class and the `getExtension` function ([link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-23ac5d986d011d0020b71866085025bdad642bd54c9c60588277bb3b5860b370L122-R128), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-56b6091c5dfa529ced7473f9d74917f33e8ecfb6039a7ad9217fa9f3ada7a471L52-R59))
*  Update the `containsSumiContributes` method in the `ext.host.ts` file to use the `sumiContributes` property instead of the `kaitianContributes` property ([link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-62990366c95d4e68747b72694a7f1f31eeeccd44aa455911ff445b0b3375d135L350-R350))
*  Update the comments and the localization values that refer to the `kaitianContributes` property in the `ext.host.ts`, `merge-contributes.ts`, `en-US.lang.ts`, and `zh-CN.lang.ts` files ([link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-62990366c95d4e68747b72694a7f1f31eeeccd44aa455911ff445b0b3375d135L512-R512), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-a8a97067eabc6c14292002497312bf4f68d3354885f3af9b71f1622de3949b1dL2-R2), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-c20faf7a0479fc71b0143e07913ac8eb35f22537d41bf74b12fc3aec686127d0L2-R2), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-77fd021913ae53d95584cf31e7a51658aca21ae8f865adf8d0e97b8205d04ac3L2-R2))
*  Update the mock extension values that use the `kaitianContributes` property in the test files ([link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-bdb1f85cb61f115d10f4d33e41403e8e38bdf6b43a4f1d81a33dea2417c8e460L55-R55), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-1756b2ee05d53914deb4cca2cf5d3052151039aa92dbdd410e9e7c0b4d2e9fbdL124-R124), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-1756b2ee05d53914deb4cca2cf5d3052151039aa92dbdd410e9e7c0b4d2e9fbdL310-R310), [link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-c88fe93e73be2eebb160a25a9aeea6dcfa93a0e5aed55068fb9943a20fe6ea01L30-R30))
*  Remove an empty line from the `extension.scanner.ts` file ([link](https://github.com/opensumi/core/pull/2664/files?diff=unified&w=0#diff-23ac5d986d011d0020b71866085025bdad642bd54c9c60588277bb3b5860b370L9))

支持 `sumiContributes` 贡献点同时兼容 `kaitianContributes` 写法，当仅有 `kaitianContributes` 存在时，使用 `kaitianContributes`, 其他情况一律采用 `sumiContributes`

重要修改见第一个提交

### <samp>🤖 Generated by Copilot at 74babfe</samp>

This pull request renames the `kaitianContributes` property to `sumiContributes` in various files related to the extension framework and contribution points. This is to unify the naming convention and avoid confusion with the deprecated `kaitian.js` entry point. The pull request also adds some compatibility checks and code style improvements for the extension contribution points.

### CODE REVIEW

File: packages/core-browser/src/extensions/extensions-point.service.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export class ExtensionsPointServiceImpl implements IExtensionsSchemaService {
   private appendPropertiesFactory(kind: FrameworkKind): (points: string[], desc: IExtensionPointDescriptor) => void {
     const properties =
       kind === 'opensumi'
-        ? OpensumiExtensionPackageSchema.properties!.kaitianContributes.properties
+        ? OpensumiExtensionPackageSchema.properties!.sumiContributes.properties
```
Comment (Ricbet) [inline packages/core-browser/src/extensions/extensions-point.service.ts:30]: 这样改可能就影响了原先使用了 kaitianContributes 的插件的代码补全提示了，不过这个字段都已经 deprecated 了，没有也没关系。可以补充在 changelog 里

File: packages/i18n/src/common/contributes/zh-CN.lang.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 export const browserViews = {
-  // kaitianContributes
+  // sumiContributes
   'sumiContributes.opensumiContributes': '由此包表示的 Opensumi 扩展的所有贡献',
```
Comment (Ricbet) [inline packages/i18n/src/common/contributes/zh-CN.lang.ts:3]: 这里可以保留 kaitianContributes 的 properties，然后加个描述提示说用 sumiContributes 吧

File: packages/core-browser/src/extensions/schema/opensumiExtensionPackageSchema.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { localize } from '@opensumi/ide-core-common';
 
 export const OpensumiExtensionPackageSchema = {
   properties: {
-    kaitianContributes: {
+    sumiContributes: {
       description: localize('sumiContributes.opensumiContributes'),
```
Comment (Ricbet) [inline packages/core-browser/src/extensions/schema/opensumiExtensionPackageSchema.ts:6]: 这里可以保留 kaitianContributes 的 properties，然后加个描述提示说用 sumiContributes 吧

File: packages/core-browser/src/extensions/extensions-point.service.ts (hunk 1 above)
Comment (erha19) [reply packages/core-browser/src/extensions/extensions-point.service.ts:30]: 这里是之前修改遗留的问题，依旧没有把 kaitianContributes 改成 sumiContributes，整体上面插件不会有影响，只是可能用了新版本集成的 IDE，不会再提示 kaitianContributes 的补全信息，应该不是啥 Big Change

**Review by Ricbet — APPROVED** [review]

**Pull request merged**